src/pages/rekey-reason/rekey-reason.ts

```diff
@@ -122,4 +203,47 @@ export class RekeyReasonPage extends PracticeableBasePageComponent {
     }
   }
 
+  ipadIssueSelected() {
+    console.log('asdasdasdASSAASD');
+    this.store$.dispatch(new IpadIssueSelected(this.formGroup.controls['ipadIssue'].value));
+  }
+  ipadIssueTechFaultChanged() {
+    this.store$.dispatch(new IpadIssueTechFault(this.formGroup.controls['ipadIssueTechFault'].value === 'true'));
+  }
+  ipadIssueLostChanged() {
+    this.store$.dispatch(new IpadIssueLost(this.formGroup.controls['ipadIssueLost'].value === 'true'));
+  }
+  ipadIssueStolenChanged() {
+    this.store$.dispatch(new IpadIssueStolen(this.formGroup.controls['ipadIssueStolen'].value === 'true'));
+  }
+  ipadIssueBrokenChanged() {
+    this.store$.dispatch(new IpadIssueBroken(this.formGroup.controls['ipadIssueBroken'].value === 'true'));
+  }
+  otherSelected() {
+    this.store$.dispatch(new OtherSelected(this.formGroup.controls['otherSelected'].value));
+  }
+  otherReasonUpdatedChanged() {
+    this.store$.dispatch(new OtherReasonUpdated(this.reasonValue()));
+  }
+
+  reasonValue(): string {
+    return this.formGroup.controls['otherReasonUpdated'].value;
+  }
+
+  characterCountChanged(charactersRemaining: number) {
+    this.reasonCharsRemaining = charactersRemaining;
+  }
+  getCharacterCountText() {
+    const characterString = Math.abs(this.reasonCharsRemaining) === 1 ? 'character' : 'characters';
+    const endString = this.reasonCharsRemaining < 0 ? 'too many' : 'remaining';
+    return `You have ${Math.abs(this.reasonCharsRemaining)} ${characterString} ${endString}`;
+  }
+  charactersExceeded(): boolean {
```

Comment:
method created charactersExceeded, but not used. Could be used in line 238  for 'too many':'remaining' logic

Reply:
It's used in the HTML template :D
 <div class="character-count validation-text" [ngClass]="{'ng-invalid': charactersExceeded()}">